doc/nrf/doc_styleguide.rst

```diff
@@ -259,32 +320,34 @@ To link to functions, enums, or structs from within Doxygen itself, use the
 
 	/** @brief Event header structure.
 	 *  Use this structure with the function @ref function_name and
-         *  this structure is related to another structure, @ref structure_name.
+     *  this structure is related to another structure, @ref structure_name.
 	 */
 
 .. note::
    Linking to functions does not currently work due to `breathe issue #438`_.
 
 
-Typedefs - WIP
-==============
+Typedefs
+========
 
-#. The documentation block should follow, not precede, the documented element.
-#. The C99-style single line comment, ``//``, is not allowed, as per `Zephyr coding style`_.
+The documentation block should precede the documented element.
+This is in accordance with the `Zephyr coding style`_.
 
 .. code-block:: c
-   :caption: Typedef documentation example -- PH
-
-   TBD
-
-TBD
-==============
-
-@def, @fn should not be used for defines or functions; Zephyr seems to require this but we should be ok without this.
-Just use a @brief and let doxygen figure out what exactly you are documenting.
-
-For parameters, it is recommended to specify whether they are [in], [out], or [in,out].
-If you specify this for one parameter in a function, all others must have it as well, for consistency. *To be discussed if this should be a requirement.*
+   :caption: Typedef documentation example
+
+   /**
+    * @brief Download client asynchronous event handler.
+    *
+    * Through this callback, the application receives events, such as
+    * download of a fragment, download completion, or errors.
+    *
+    * If the callback returns a non-zero value, the download stops.
+    * To resume the download, use @ref download_client_start().
+    *
+    * @param[in] event	The event.
+    *
+    * @return Zero to continue the download, non-zero otherwise.
```

Comment:
Not consistent with the example above.

Reply:
Modified it in this file only. These are examples, after all.